jaxrs-spec/src/main/asciidoc/chapters/filters_and_interceptors/_exceptions_filters_and_interceptors.adoc

```diff
@@ -17,7 +17,8 @@
 When a filter or interceptor method throws an exception, the server
 runtime will process the exception as described in
 <<exceptions_providers_server>>. As explained in
-<<exceptionmapper>>, an application can supply exception mapping
+<<exceptionmapper>>, a JAX-RS implementation will supply a default
+exception mapping provider and an application can supply additional exception mapping
```

Comment:
Hi @chkal. Yes, good poiint. I'll make that change.

Reply:
Isn't the notion of a _default_ mapper something that can be overridden by the application developer? That is, _if_ the application defines such a mapper, the _default_ one should be inactive --it's more of an override than a re-prioritization.

Reply:
Yes, in some regard the notation of a *default mapper* already emphasizes that it can be overwritten. However, IMO there are various corner cases which needs to be considered. What about mappers deployed as part of 3rd party libraries in `WEB-INF/lib`? Do they override default mappers? And in Java SE you even deploy your JAX-RS implementation *and* 3rd party libraries via `WEB-INF/lib`. Therefore, I think that using `@Priority` for ordering is easier to reason about.

Reply:
Right, but that is already part of the spec for all providers (and exception mappers - see Section 4.4); what is being proposed here is a new _default_. Surely, this new default would need to have the "right" priority (as other default providers like MBR and MBW do) in order to comply with the existing rules.

Reply:
I cannot look up the wording in the spec right now, but IIRC section 4.4 only specifies that a priority can be used to get predictable ordering if multiple mappers have the same "distance" in the inheritance hierarchy to the relevant exception type. But I may be wrong about that.

Reply:
@chkal, from Section 4.4:
```
When choosing an exception mapping provider to map an exception, an
implementation MUST use the provider whose generic type is the nearest
superclass of the exception. If two or more exception providers are applicable,
the one with the highest priority MUST be chosen as described in Section 4.1.3.
```
So I think the default ExceptionMapper should implement ExceptionMapper<Throwable> and have priority Integer.MAX_VALUE, i.e., minimal priority.

Reply:
I finally found some time to check the exact wording regarding `@Priority` for providers. In section 4.1.3 the spec states:

```
Application-supplied providers enable developers to extend and customize the 
JAX-RS runtime. Therefore, an application-supplied provider MUST always be 
preferred over a pre-packaged one if a single one is required. [...]
```

So I agree with @spericas that there is no need to explicitly describe priority ordering in the exception mapper section. 

> So I think the default ExceptionMapper should implement ExceptionMapper<Throwable> and have priority Integer.MAX_VALUE, i.e., minimal priority.

I like the idea that even default providers have a priority (like `Integer.MAX_VALUE`) because it makes the resolving algorithm simpler. In this case we don't need to distinguish between "application-supplied" and "pre-packaged" providers because all that matters is the priority.

However, I guess this is out of scope for the PR. So let us get this one merged first.